### PR TITLE
feat: filtering channels by funding wallet

### PR DESCRIPTION
**Pull request by markettes**

### CODE REVIEW

File: src/Data/Models/Channel.cs [hunk 1]
```diff
@@ -44,31 +46,43 @@ public enum ChannelStatus
...
+        {
+            get
+            {
+                if (ChannelOperationRequests != null && ChannelOperationRequests.Count > 0)
+                    return ChannelOperationRequests.FirstOrDefault().Wallet.Id;
+                else
+                    return null;
+            }
+        }
+
```
Comment (RodriFS) [inline src/Data/Models/Channel.cs:76]: would this work? 🤔
```
ChannelOperationRequests?.FirstOrDefault()?.Wallet?.Id;
```

File: src/Pages/Channels.razor [hunk 2]
```diff
@@ -692,6 +715,19 @@
...
+        {
+            return true;
+        }
+        if (itemValue == null) return false;
+        if ((int)itemValue == (int)searchValue)
+        {
+            return true;
+        }
+        return false;
+    }
```
Comment (RodriFS) [inline src/Pages/Channels.razor:730]: This should be the same, right?
```
return searchValue == null || (int)searchValue == 0 || (int)itemValue == (int)searchValue
```

**Review by RodriFS — APPROVED** [review]

File: src/Data/Models/Channel.cs (hunk 1 above)
Comment (markettes) [reply src/Data/Models/Channel.cs:76]: Its true, good point!

File: src/Pages/Channels.razor [hunk 3]
```diff
@@ -93,7 +92,27 @@
...
+                        </Select>
+                    </FilterTemplate>
+                    <DisplayTemplate>
+                        @{
+                            Wallet wallet = null;
+                            if (context.OpenedWithId != null)
+                            {
+                                wallet = Task.Run(() => WalletRepository.GetById((int)context.OpenedWithId)).Result;
+                            }
+                            @(wallet == null ? "Unknown" : wallet.Name)
```
Comment (Jossec101) [inline src/Pages/Channels.razor:112]: I wonder how much will this make the channels view slower, its async but maybe slows down, we already have the problem with the balance, worth researching

File: src/Pages/Channels.razor [hunk 4]
```diff
@@ -692,6 +715,11 @@
         return true;
     }
 
+    private bool OnWalletFilter(object itemValue, object searchValue)
+    {
+        if (itemValue == null) return false;
```
Comment (RodriFS) [inline src/Pages/Channels.razor:720]: Do you really need this line?

**Review by RodriFS — APPROVED** [review]

File: src/Pages/Channels.razor (hunk 4 above)
Comment (markettes) [reply src/Pages/Channels.razor:720]: yes, because on some cases you get null on the itemvalue

**Pull request merged**